ts/navigation/MainNavigator.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from "react";
+import ROUTES from "./routes";
+
+import { Icon } from "native-base";
+import { TabBarBottom, TabNavigator } from "react-navigation";
+import MessagesScreen from "../screens/main/MessagesScreen";
+import ProfileScreen from "../screens/main/ProfileScreen";
+import WalletNavigator from "./WalletNavigator";
+
+type RouteType = keyof typeof ROUTES;
+
+type RouteIconMap = { [key in RouteType]?: string };
+
+const ROUTE_ICON: RouteIconMap = {
+  MAIN_MESSAGES: "mail",
+  WALLET_HOME: "wallet",
+  DOCUMENTS_HOME: "document",
+  PREFERENCES_HOME: "cog",
+  MAIN_PROFILE: "user"
+};
+
+const getIcon = (routeName: RouteType): string => {
+  const unknownIcon = "???";
+
+  for (const route in ROUTE_ICON) {
```

Comment:
I don't understand this logic, why not `ROUTE_ICON[routeName]` ?

Reply:
I honestly don't know why that was being done that way. Updated to `ROUTE_ICON[routeName]` (that required introducing a type including all and only the ROUTES having icons (to avoid undefined values))